pkg/management/component.go

```diff
@@ -827,7 +841,25 @@ func (c *component) getUserRolesAsMap(ctx context.Context, realmName, userID str
 		c.logger.Info(ctx, "msg", "Failed to get user roles", "realm", realmName, "user")
 		return nil, err
 	}
-	return c.rolesToMap(roles), nil
+	userRoles, err := c.getRolesWithAttributes(accessToken, realmName, roles)
+	if err != nil {
+		c.logger.Info(ctx, "msg", "Failed to get user roles", "realm", realmName, "user")
+		return nil, err
+	}
+
+	return c.rolesToMap(userRoles), nil
+}
+
+func (c *component) getRolesWithAttributes(accessToken string, realmName string, roles []kc.RoleRepresentation) ([]kc.RoleRepresentation, error) {
```

Comment:
This way of retrieving Roles attributes is only used when we retrieve role mapping on a user. Keycloak does not offer to retrieve role-mapping directly with attributes. We may consider to extend the keycloak api to offer to retrieve in one call the role-mappings and their attributes. The question is do we will have user having a lot of roles ? Is there a need of scalability ?